dotCMS/src/main/java/com/dotmarketing/portlets/rules/parameter/display/Input.java

```diff
@@ -32,5 +38,45 @@ public T getDataType() {
     public void checkValid(String value)  throws InvalidRuleParameterException{
     	return;
     }
-}
 
+    /**
+     * Utility type used to correctly read immutable object from JSON representation.
+     *
+     * @deprecated Do not use this type directly, it exists only for the <em>Jackson</em>-binding
+     * infrastructure
+     */
+    @Deprecated
+    @JsonDeserialize
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
+    @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.NONE)
+    static final class Json {
+
+        @javax.annotation.Nullable
+        String id;
+        @javax.annotation.Nullable
+        DataType dataType;
+
+        @JsonProperty("id")
+        public void setKey(@Nullable String id) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'id' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4925354)